torch/autograd/functional.py

```diff
@@ -342,18 +341,37 @@ def jvp(func, inputs, v=None, create_graph=False, strict=False):
                                "the user-provided function is a single Tensor "
                                "with a single element.")
 
-    outputs = func(*inputs)
-    is_outputs_tuple, outputs = _as_tuple(outputs, "outputs of the user-provided function", "jvp")
-    _check_requires_grad(outputs, "outputs", strict=strict)
-    # The backward is linear so the value of grad_outputs is not important as
-    # it won't appear in the double backward graph. We only need to ensure that
-    # it does not contain inf or nan.
-    grad_outputs = tuple(torch.zeros_like(out, requires_grad=True) for out in outputs)
+    if fw_mode:
+        if v is None:
+            v = (torch.ones_like(inputs[0]),)
+
+        with fwAD.dual_level():
+            inputs_dual = [fwAD.make_dual(el_inp, el_v) for el_inp, el_v in zip(inputs, v)]
 
-    grad_inputs = _autograd_grad(outputs, inputs, grad_outputs, create_graph=True)
-    _check_requires_grad(grad_inputs, "grad_inputs", strict=strict)
+            outputs_dual = func(*inputs_dual)
+            is_outputs_tuple, outputs_dual = _as_tuple(outputs_dual, "outputs of the user-provided function", "jvp")
 
-    grad_res = _autograd_grad(grad_inputs, grad_outputs, v, create_graph=create_graph)
+            outputs, grad_res = zip(*[fwAD.unpack_dual(o_dual) for o_dual in outputs_dual])
+
+        if strict:
+            for i, g in enumerate(grad_res):
+                if g is None:
+                    raise RuntimeError("The output of the user-provided function is independent of "
+                                       "input {}. This is not allowed in strict mode.".format(i))
+
+    else:
+        outputs = func(*inputs)
+        is_outputs_tuple, outputs = _as_tuple(outputs, "outputs of the user-provided function", "jvp")
+        _check_requires_grad(outputs, "outputs", strict=strict)
+        # The backward is linear so the value of grad_outputs is not important as
+        # it won't appear in the double backward graph. We only need to ensure that
+        # it does not contain inf or nan.
+        grad_outputs = tuple(torch.zeros_like(out, requires_grad=True) for out in outputs)
+
+        grad_inputs = _autograd_grad(outputs, inputs, grad_outputs, create_graph=True)
+        _check_requires_grad(grad_inputs, "grad_inputs", strict=strict)
+
+        grad_res = _autograd_grad(grad_inputs, grad_outputs, v, create_graph=create_graph)
 
     jvp = _fill_in_zeros(grad_res, outputs, strict, create_graph, "back_trick")
```

Comment:
By "back_trick", do we mean the "double backward trick"? This is not necessarily the double backward trick anymore, especially if we use forward-mode AD, right?